### PR TITLE
OIC client mem usage fix [v2]

### PR DESCRIPTION
**Pull request by glima**

### CODE REVIEW

**Comment by glima** [comment]
@otaviobp, are you working on those fixes? Do you have an ETA?

File: src/lib/comms/sol-oic-client.c [hunk 1]
```diff
@@ -1126,31 +1160,31 @@ _poll_resource(void *data)
...
-        .client = client,
-        .cb = callback,
-        .data = data,
-        .res = res
-    }, sizeof(*ctx));
+            .base.client = client,
+            .cb = callback,
+            .base.data = data,
+            .res = res
+        }, sizeof(*ctx));
```
Comment (glima) [inline src/lib/comms/sol-oic-client.c]: I forgot to leave out these uncrustify v60 vs v61 issues: I'll fix them when merging.

**Comment by glima** [comment]
I'll be commenting out OIC bindings till they get fixed, I guess.

**Comment by glima** [comment]
Changes since v1:

File: src/lib/common/include/sol-log.h [hunk 2]
```diff
@@ -268,6 +310,22 @@ extern "C" {
...
+ * integer @c var satisfies the expression @c exp. Additionally, it
+ * sets the @c errno variable to @a err.
+ *
+ * @param var Integer to be check
+ * @param exp Safety-check expression
+ * @param err @c errno value to set
+ * @param label @c goto label
+ */
+#define SOL_INT_CHECK_GOTO_ERRNO(var, exp, err, label) \
+    SOL_INT_CHECK_GOTO_IMPL_ERRNO(var, exp, label)
```
Comment (otaviobp) [inline src/lib/common/include/sol-log.h]: err param is missing here

File: src/lib/comms/sol-oic-client.c [hunk 3]
```diff
@@ -1085,40 +1122,52 @@ sol_oic_client_request_get_writer(struct sol_oic_request *request)
...
     return _resource_request((struct sol_oic_client_request *)request, client,
         callback, (void *)callback_data);
 }
 
+SOL_API void
+sol_oic_pending_cancel(struct sol_oic_pending *pending)
+{
+    struct ctx *ctx = (struct ctx *)pending;
+
+    if (!ctx->req || !ctx->server)
```
Comment (otaviobp) [inline src/lib/comms/sol-oic-client.c]: check if ctx is NULL here.

File: src/lib/common/include/sol-log.h (hunk 2 above)
Comment (glima) [inline src/lib/common/include/sol-log.h]: Ouch, thanks

File: src/lib/comms/sol-oic-client.c (hunk 3 above)
Comment (glima) [inline src/lib/comms/sol-oic-client.c]: Thanks.

**Comment by otaviobp** [comment]
lgtm

**Comment by glima** [comment]
alright, thanks. will wait a little bit more and merge at the end of the day

**Comment by glima** [comment]
Merged with due fixes

**Comment by glima** [comment]
@gabrielschulhof, would you please take a look at oic js bindings in order to enable them back? it would take me a couple of time to figure that out but for you it should be a question of minutes.

**Pull request closed without merging**